src/settings/time.tsx

```diff
@@ -46,6 +48,15 @@ export class Time implements TimeDate {
     }
   }
 
+  prepareHoursForModalWindow() {
+    const newHours: number[] = [];
+
+    for (let i = initialHourForModalWindow; i <= maxHourForModalWindow; ++i) {
```

Comment:
try using `reduce` instead of `for` in such cases in the future

Reply:
rewrote